symengine/pow.cpp

```diff
@@ -311,29 +323,52 @@ RCP<const Basic> Log::create(const RCP<const Basic> &a) const
 
 RCP<const Basic> log(const RCP<const Basic> &arg)
 {
-    if (eq(*arg, *zero)) {
-        throw NotImplementedError(
-            "log(0) is complex infinity. Yet to be implemented");
-    }
+    if (eq(*arg, *zero))
+        return ComplexInf;
     if (eq(*arg, *one))
         return zero;
     if (eq(*arg, *E))
         return one;
+    if (eq(*arg, *Inf))
```

Comment:
@isuruf is this something like what you intended for ?
template <class T>
class EvaluateInf : public Evaluate
{
}
inside infty.cpp